Sources/Overload/OvAnalytics/src/OvAnalytics/Profiling/Profiler.cpp

```diff
@@ -74,8 +74,19 @@ void OvAnalytics::Profiling::Profiler::Save(OvAnalytics::Profiling::ProfilerSpy&
 	__SAVE_MUTEX.lock();
 
 	/* Check if this thread is already registered */
-	if (std::find(__WORKING_THREADS.begin(), __WORKING_THREADS.end(), std::this_thread::get_id()) == __WORKING_THREADS.end())
-		__WORKING_THREADS.push_back(std::this_thread::get_id());
+	{
```

Comment:
What's the purpose of this change?

Reply:
std::thread::id is not hashable by C++ standard. That means that std::find does not work except if the implementation doesn't follow the standard (like MSVC).

Reply:
> I'm not sure I understand why we would include some `Makefile` in the repository? Aren't these supposed to get generated by Premake?
> 
> Also, what is the purpose of this PR? By the title, it seems like this is a step towards cross-platform compatibility, but this should't get merged to develop if the purpose of these changes isn't fully met (full cross-platform compatibility).
> 
> I would split this one in multiple PRs:
> 
>     * one should focus on typos and specifying C++ standards in our premake files
> 
>     * one should expand our premake files to make sure they are portable and working well with Windows & Linux

I understand. This however is something I can not achieve due to lack of knowledge. However I can ease the work for someone who actually can.

The Makefiles left in there were an accident. I should add Makefile to the .gitignore

I will consider spliting this pr into multiple prs.